apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -367,6 +367,62 @@ Services.obs.addObserver(function(document) {
       return request;
     },
 
+    delete: function(id) {
+      var request = Services.DOMRequest.createRequest(window);
+      var messageToDeleteIds = Array.isArray(id) ? id : [id];
+
+      getStorage().then((storage) => {
+        var messagesToDelete = [];
+        for (var thread of storage.threads.values()) {
+          messageToDeleteIds.forEach((id) => {
+            var messageToDeleteIndex = thread.messages.findIndex((message) => {
+              return message.id === id;
+            });
+
+            if (messageToDeleteIndex >= 0) {
+              messagesToDelete.push({
+                thread: thread,
+                messageIndex: messageToDeleteIndex
+              });
+            }
+          });
+
+          // All messages found.
+          if (messagesToDelete.length === messageToDeleteIds.length) {
+            break;
+          }
+        }
+
+        if (messagesToDelete.length) {
+          var deletedThreadIds = [];
+          messagesToDelete.forEach((messageToDeleteInfo) => {
+            messageToDeleteInfo.thread.messages.splice(
+              messageToDeleteInfo.messageIndex, 1
+            );
```

Comment:
I don't think this works reliably; once you splice once the indexes are not correct anymore.
This could work if you order messageToDelete in descending order though.

but all this would be a lot less crazy if we simply had a messages map... because each message have a threadId property, we can find the thread easily; and thread.messages would simply be an array of messageIds.